zebra-chain/src/transaction/unmined.rs

```diff
@@ -174,55 +187,117 @@ pub struct UnminedTx {
 
     /// The size in bytes of the serialized transaction data
     pub size: usize,
+
+    /// The cost in bytes of the transaction, as defined in [ZIP-401].
+    ///
+    /// A reflection of the work done by the network in processing them (proof
+    /// and signature verification; networking overheads; size of in-memory data
+    /// structures).
+    ///
+    /// [ZIP-401]: https://zips.z.cash/zip-0401
+    pub cost: u32,
+}
+
+impl fmt::Display for UnminedTx {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("UnminedTx")
+            .field("transaction", &self.transaction)
+            .field("serialized_size", &self.size)
+            .finish()
+    }
 }
 
 // Each of these conversions is implemented slightly differently,
 // to avoid cloning the transaction where possible.
 
 impl From<Transaction> for UnminedTx {
     fn from(transaction: Transaction) -> Self {
+        let size = transaction
+            .zcash_serialized_size()
+            .expect("all transactions have a size");
```

Comment:
Nitpick: There is one exception to this rule - some generated transactions are structurally invalid, so they won't serialize.

Can you also make this change for every other `size` variable?

```suggestion
            .expect("unexpected serialization failure: all structurally valid transactions have a size");
```

Reply:
Resolved in a9e02fae